pytorch_lightning/trainer/trainer.py

```diff
@@ -750,10 +750,6 @@ def run_evaluation(self, max_batches=None, on_epoch=False):
         # hook
         self.evaluation_loop.on_evaluation_epoch_end()
 
-        # update epoch-level lr_schedulers
-        if on_epoch:
-            self.optimizer_connector.update_learning_rates(interval='epoch')
```

Comment:
the reason I added this to update learning_rate inorder to have the correct state for lr_schedulers since they are save in `on_validation_end` hook in the next line. `on_epoch=True` only when the train loop is finished and should not be a problem when val_check_interval < 1. since in such a case val loop is called within the training loop up there where on_epoch=False (default) so it shouldn't be updating this learning rate.

Reply:
okay, now I better understand why it is there. I originally hesitated to remove it because I thought that there must have been a good reason why you put it there in the first place :]